### PR TITLE
add amazon linux 2 support

### PR DESCRIPTION
**Pull request by mears148**
Adds Amazon Linux 2 support based off of the baseline values of the Amazon Linux 2 AMI.

### CODE REVIEW

**Comment by mears148** [comment]
Not sure how to handle the travis tests. Any guidance on this?

File: manifests/init.pp [hunk 1]
```diff
@@ -241,8 +241,8 @@
 ) {
 
   # Fail on unsupported platforms
-  if $facts['os']['family'] == 'RedHat' and !($facts['os']['release']['major'] in ['5','6','7','8']) {
-    fail("osfamily RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 5, 6, 7 or 8")
+  if $facts['os']['name'] =~ 'RedHat' and !($facts['os']['release']['major'] in ['2','5','6','7','8']) {
```
Comment (ghoneycutt) [inline manifests/init.pp:244]: Why the change from `==` to `=~` ? I don't think that is necessary.

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

**Comment by ghoneycutt** [comment]
@mears148 awesome PR! thank you!

File: manifests/init.pp (hunk 1 above)
Comment (mears148) [reply manifests/init.pp:244]: That is a mistake. I was going to try and make a regex to resolve the os name but I thought that might create new problems in other areas so I left it alone. It can be changed back to == with no issues.

**Comment by mears148** [comment]
@ghoneycutt Please let me know if you need anything else done to get this merged to the master branch. Not sure what to do about the travis tests. Thanks.

**Comment by ghoneycutt** [comment]
@mears148 Created https://github.com/ghoneycutt/puppet-module-pam/pull/228 once it gets merged you should be able to rebase and force push to this branch to retrigger testing.

**Comment by ghoneycutt** [comment]
I went ahead and rebased, opening https://github.com/ghoneycutt/puppet-module-pam/pull/229
